src/lib.rs

```diff
@@ -46,10 +48,10 @@ use single_instruction_test::stack;
 #[cfg(feature = "single_instruction_test")]
 pub use zkevm_opcode_defs;
 
-pub trait World {
+pub trait World<T> {
```

Comment:
The parameter is necessary because dynamic dispatch breaks otherwise. It may be possible to statically dispatch World now, though.

Also, there are probably other solutions. I added the parameter to get something working without analyzing the problem properly.

Reply:
Without this parameter the world would have to have program caches for many different tracer types.